packages/snap/src/index.ts

```diff
@@ -190,7 +199,16 @@ export const onRpcRequest = async ({
           createProofRequest.query,
           issuerDid,
         )
-      ).filter((cred) => cred.credentialSubject.id === identityStorage.did);
+      ).filter((cred) => {
+        const splittedCredSubjId = String(cred.credentialSubject.id).split(':');
```

Comment:
`split(':')` is spread all over the place. Consider encapsulating this logic into a function like `parseDidV2(...)`